### ISSUE TITLE
compilation on gles produces warnings

### ISSUE THREAD

**Issue by kristianlm** [AlexCharlton/chicken-opengl-glew#1]
Hi Alex,

I was experimenting with the opengl-glew egg on my Android tablet, and ran into some strange problems with float-point sizes. After some failed debugging, I found lots (over 100) of warnings like this in my compilation log (when compiling this egg for target):

```
opengl-glew.c: In function ‘stub2440’:
opengl-glew.c:2713:1: warning: implicit declaration of function ‘glUniform2f’ [-Wimplicit-function-declaration]
 glUniform2f(a0 , a1 , a2);
 ^
```

And this patch fixes the problem:

``` patch
modified   opengl-glew.scm
@@ -3,7 +3,7 @@
 (use bind)

 (cond-expand
- (gles #f)
+ (gles (foreign-declare "#include <GLES3/gl3.h>"))
  (else (foreign-declare "#include <GL/glew.h>")))
```

And I suppose that the warning make sense, since we're missing an OpenGL header file when compiling for GLES, and the "implicit declarations" end up being double precision.

Note that the warnings still appear when compiling for host, but that shouldn't be a problem since it just affects the host-part of cross-chicken.

Forking and PR for this one seemed a little overkill, I hope that's ok.

**Comment by AlexCharlton** [comment]
Thanks for catching that, Kristian! My ES testing has been... light. I take it you've been doing some experiments on Android? Not something I've tried myself, but it sounds fun :)

Applied your patch in df25150f60168396451350575649bc89cf7f9243

**Comment by kristianlm** [comment]
Yes, and it's a lot of fun! If you're interested, I've attached an Android app with a mandelbrot experiment of mine. It's using your gl-utils/gl-math eggs, but no glls (it's not really ready for ES yet). Your eggs are really nice to work with! I like your optional RESULT parameters in your gl-math egg. If I get around to it, I'll use that for my glm egg.

[SDLActivity-debug.apk.zip](https://github.com/AlexCharlton/chicken-opengl-glew/files/78434/SDLActivity-debug.apk.zip). It's zipped because github says so... I used my [Android template project](https://github.com/chicken-mobile/chicken-android-template) to make that.
